### PR TITLE
Add --banner config option

### PR DESCRIPTION
**Pull request by richsalz**
Use it in the automated workflows.

Where do I update things so that the FIPS config/build stuff adds this flag?  Ping @t8m

### CODE REVIEW

**Comment by levitte** [comment]
Not sure I like `no-banner`...  makes it look like a building option like, say, `no-fips`.  I'm pretty sure you now have a macro `OPENSSL_NO_BANNER`, among others :wink:

**Comment by richsalz** [comment]
Based on your comment, I changed it to `--banner="text..." which I think is better.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
         sudo apt-get -yq --no-install-suggests --no-install-recommends --force-yes install unifdef
     - uses: actions/checkout@v2
     - name: config
-      run: ./config --strict-warnings enable-fips && perl configdata.pm --dump
+      run: ./config --baner=Configured --strict-warnings enable-fips && perl configdata.pm --dump
```
Comment (levitte) [inline .github/workflows/ci.yml:24]: You might want to check your spelling :wink:

File: Configure [hunk 2]
```diff
@@ -2812,23 +2835,7 @@ or position independent code, please let us know (but please first make sure
...
-***   and include the output from the following command:           ***
-***                                                                ***
-***       perl configdata.pm --dump                                ***
-***                                                                ***
-***   (If you are new to OpenSSL, you might want to consult the    ***
-***   'Troubleshooting' section in the INSTALL.md file first)      ***
-***                                                                ***
-**********************************************************************
-EOF
+print $banner;
```
Comment (levitte) [inline Configure:2838]: A detail, you might want to consider this:

``` perl
$banner .= "\n" unless $banner =~ m|\R$|;
print $banner;
```

This is to make sure that, for example, `--banner=Blah-di-blah` doesn't give you a weird looking prompt on Unixly environments:

``` console
$ ./Configure --banner=Blah-di-blah
...
Blah-di-blah$ [blinking cursor here]
```

**Review by levitte — CHANGES_REQUESTED** [review]

**Comment by richsalz** [comment]
i had already fixed the issues you found, @levitte  Taken out of draft.

**Review by levitte — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Comment by levitte** [comment]
Merged after a slight edit of the commit message (modified to say `--banner`)

d0364dcc42b151cfc08d860efb15cd48d87302c6 Add --banner config option

**Comment by richsalz** [comment]
> Merged after a slight edit of the commit message (modified to say --banner)

Oh yeah, thanks.  I realized that the message had become out of date, and then forgot to do anything about it :)

**Pull request closed without merging**